python/ray/tests/test_basic_3.py

```diff
@@ -156,10 +157,17 @@ def f():
         return os.getpid(), g.remote()
 
     nested = ray.get([f.remote() for _ in range(10)])
+    pids = []
     while nested:
         pid, g_id = nested.pop(0)
-        ray.get(g_id)
-        del g_id
+        pids.append(pid)
+        assert ray.get(g_id) == 0
+        g_id = None
+
+    nested = None
+    # Force dereference all objects, so workers can exit.
```

Comment:
What is the main reason for this change?

Reply:
Simplified the change to just running GC. Basically without explicit GC, there would be leftover ObjectRefs in workers, preventing workers from exiting. A possible cause is that using the list of objects refs for `ray.get()`, compared to using ObjectRef individually, has different behavior under Python GC which affects Ray ref counting. But I don't have more insights into the GC behavior yet.